include/aws/io/socket.h

```diff
@@ -30,6 +30,13 @@ enum aws_socket_type {
     AWS_SOCKET_DGRAM,
 };
 
+#ifdef AWS_OS_WINDOWS
+#    define AWS_NETWORK_INTERFACE_MAX_LEN 16
+#else
+#    include <net/if.h>
+#    define AWS_NETWORK_INTERFACE_MAX_LEN IFNAMSIZ
```

Comment:
`AWS_NETWORK_INTERFACE_NAME_MAX` instead of ~AWS_NETWORK_INTERFACE_MAX_LEN~. It's confusing to use "LEN" to mean a string size that includes the `\0`, when `strlen()` doesn't include the `\0` in it.

This aligns with some [POSIX](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/limits.h.html) names like `PATH_MAX` and `NAME_MAX`

I'm sure there are examples of `MAX_LEN` in our codebase, but 🤷‍♀️